File: backend/src/main/java/net/tape/model/ScheduleSlot.java
package net.tape.model;

import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

/** Domain model — framework-free POJO. */
@Getter
@Setter
public class ScheduleSlot implements Serializable {
    private Long id;
    private Integer dayOfWeek;
    private String timeEt;
    private Long showId;
    private String showName;
    private String showSlug;
    private String hostsLabel;
    private boolean live;
    private int sort;
}

File: web/src/api/components/schemas/ScheduleSlot.ts
import { Option } from 'fp-ts/lib/Option';
import { Int, string, boolean, type } from 'io-ts';
import { optionFromNullable } from 'io-ts-types/lib/optionFromNullable';

export type ScheduleSlot = {
	id: Option<Int>;
	dayOfWeek: Option<Int>;
	timeEt: Option<string>;
	showId: Option<Int>;
	showName: Option<string>;
	showSlug: Option<string>;
	hostsLabel: Option<string>;
	live: Option<boolean>;
	sort: Option<Int>;
};
export const ScheduleSlotIO = type(
	{
		id: optionFromNullable(Int),
		dayOfWeek: optionFromNullable(Int),
		timeEt: optionFromNullable(string),
		showId: optionFromNullable(Int),
		showName: optionFromNullable(string),
		showSlug: optionFromNullable(string),
		hostsLabel: optionFromNullable(string),
		live: optionFromNullable(boolean),
		sort: optionFromNullable(Int),
	},
	'ScheduleSlot',
);
